src/components/shape/shape.js

```diff
@@ -0,0 +1,56 @@
+var CANNON = require('cannon');
+
+var Shape = {
+  schema: {
+    shape: {default: 'box', oneOf: ['box', 'sphere', 'cylinder']},
+    radius: {type: 'number', default: 1},
+    offset: {type: 'vec3', default: {x: 0, y: 0, z: 0}},
+    halfExtents: {type: 'vec3', default: {x: 1, y: 1, z: 1}},
+    orientation: {type: 'vec4', default: {x: 0, y: 0, z: 0, w: 1}},
+    radiusTop: {type: 'number', default: 1},
+    radiusBottom: {type: 'number', default: 1},
+    height: {type: 'number', default: 1},
+    numSegments: {type: 'int', default: 8}
```

Comment:
let's prefix these properties:

```
sphereRadius
boxHalfExtents
cylinderRadiusTop
...
```

Reply:
I would prefer not to- I really like that this exactly matches what those properties are called in threejs. (Selfishly I am also writing some tooling upstream of this that I don't want to have to constantly try to remember what the properties are supposed to be called). I would argue that if we want to be more explicit, we should add the other shape components back in (`sphere-shape`, `box-shape`, `cylinder-shape`)

Reply:
What about adding a `updateSchema` handler a la `geometry` that makes the right properties available for the selected shape? That may provide the same clarity of purpose as prefixing

Reply:
Ok, I'm fine with keeping the names then. But even easier than `updateSchema`, we can do this:

```
radiusTop: { type: 'number', default: 1, if: {shape: ['cylinder']} }
...
```